include/cuco/static_map.cuh

```diff
@@ -197,6 +197,8 @@ class static_map {
    * this sentinel value.
    * @note If a non-default CUDA stream is provided, the caller is responsible for synchronizing the
    * stream before the object is first used.
+   * @note Static extent should not use this overload since it will convert all extents into a
+   * runtime constant.
```

Comment:
Nitpick: I would change the note to "This overload will convert compile-time extents to runtime constants which might lead to performance regressions."